File: php/src/Series.php
<?php

declare(strict_types=1);

namespace MisarBlog;

/**
 * A series/collection of articles, per the `Series` schema.
 */
final class Series
{
    public function __construct(
        public readonly ?string $id = null,
        public readonly ?string $slug = null,
        public readonly ?string $title = null,
        public readonly ?string $description = null,
        public readonly ?int $articleCount = null,
        public readonly ?string $createdAt = null,
    ) {}

    /**
     * @param array<string,mixed> $json
     */
    public static function from(array $json): self
    {
        return new self(
            id: $json['id'] ?? null,
            slug: $json['slug'] ?? null,
            title: $json['title'] ?? null,
            description: $json['description'] ?? null,
            articleCount: isset($json['article_count']) ? (int) $json['article_count'] : null,
            createdAt: $json['created_at'] ?? null,
        );
    }
}

File: java/src/main/java/blog/misar/sdk/models/Series.java
package blog.misar.sdk.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A series (collection) of articles.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Series {
    public String id;
    public String slug;
    public String title;
    public String description;

    @JsonProperty("article_count")
    public int articleCount;

    @JsonProperty("created_at")
    public String createdAt;

    @Override
    public String toString() {
        return "Series{slug=" + slug + ", title=" + title + ", articleCount=" + articleCount + "}";
    }
}
